test/compile.jl

```diff
@@ -474,4 +524,61 @@ let module_name = string("a",randstring())
     rm(file_name)
 end
 
+# Issue #19960
+let
+    # ideally this would test with workers on a remote host that does not have access to the master node filesystem for loading
+    # can simulate this for local workers by using relative load paths on master node that are not valid on workers
+    # so addprocs before changing directory to temp directory, otherwise workers will inherit temp working directory
+
+    test_workers = addprocs(1)
+
+    temp_path = mktempdir()
+    cd(temp_path)
+    load_path = mktempdir(temp_path)
+    load_cache_path = mktempdir(temp_path)
+    unshift!(LOAD_PATH, basename(load_path))
+    unshift!(Base.LOAD_CACHE_PATH, basename(load_cache_path))
+
+    ModuleA = :Issue19960A
+    ModuleB = :Issue19960B
+
+    write(joinpath(load_path, "$ModuleA.jl"),
+        """
+        __precompile__(true)
+        module $ModuleA
+            export f
+            f() = myid()
+        end
+        """)
+
+    write(joinpath(load_path, "$ModuleB.jl"),
+        """
+        __precompile__(true)
+        module $ModuleB
+            using $ModuleA
+            export g
+            g() = f()
+        end
+        """)
+
+    try
+        @eval using $ModuleB
+
+        @test g() == 1
+        for wid in test_workers
+            @test remotecall_fetch(g, wid) == wid
+        end
+
+    catch
+        @test false
```

Comment:
something went wrong with the rebase as well, when you do `git rebase -i origin/master` you should remove all the commits that aren't yours from the list that shows up, and when you're happy with the local state of the branch (just your commits, on top of latest master) you can push to the `glp/remote_loading` branch of your fork with `--force`